src/ray/protobuf/node_manager.proto

```diff
@@ -19,6 +19,17 @@ package ray.rpc;
 import "src/ray/protobuf/common.proto";
 import "src/ray/protobuf/gcs.proto";
 
+message ReportWorkerBacklogRequest {
+  // Unique id of the worker that's reporting the backlog
+  bytes worker_id = 1;
+  // TaskSpec containing the requested resources.
+  TaskSpec resource_spec = 2;
+  int64 backlog_size = 3;
+}
```

Comment:
Since backlog size doesn't include tasks with pending lease requests, so whenever we send a lease request, backlog size changes, that's why I call `ReportWorkerBacklogIfNeeded` after requesting lease.

In terms of simplicity, this is actually simpler I think: no timer, no need to figure out the report frequency. Also this matches better with current behavior (i.g. backlog report as part of lease request) so less surprise.

I think the key here is that we decouple lease request and backlog report request so we can always change how we report in the future. @wuisawesome thoughts?

Reply:
Actually this makes me think of an important edge case, where a stale backlog can cause extra GPU nodes to get allocated.

Suppose you have two GPU requests and both are leased at the same time. If the autoscaler gets a snapshot of "two pending GPU requests" and "backlog size 1", then it could launch 3 GPU machines.

Reply:
When raylets reports to autoscaler, it reports both the backlog and pending lease requests. So if backlog also includes pending lease requests, then we will overcount. @wuisawesome correct me if I'm wrong.

Reply:
Ah gotcha.

Hmm this actually seems quite complicated if parallel pending lease requests go to different nodes. Not sure if we can get an exact backlog count in that scenario.